pages/docs/concepts/asyncapi-document/resuable-parts.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Reusable parts
+weight: 249
+---
+
+Reusable parts in AsyncAPI provide flexibility, modularity, and code reusability. Using reusable parts makes it easier to generate code and validate the specifications. You can reuse specific document sections such as Messages or schema definitions.
+
+Reusable parts allow you to split up the AsyncAPI document into many files and reference them using the Reference Object ($ref). You can use the $ref keyword to reference the same or another local file or external URL.
+
+## Same file
+
+You can use the ref keyword to reference a component within the same document. The image below defines referencing a component within the same document.
+
+```mermaid
+graph LR
+style A fill:#47BCEE, stroke:#333, stroke-width:2px;
+style C fill:#47BCEE, stroke:#333, stroke-width:2px;
+style D fill:#47BCEE, stroke:#333, stroke-width:2px;
+style B fill:#47BCEE, stroke:#333, stroke-width:2px;
+
+A[MyMessageSchema]
+B[components]
+C[myChannel]
+D[$ref keyword]
+
+C -->|publishes| D
+D -->|references| B
+B -->|references| A
+```
+
+In the example below, you define a component called MyMessageSchema under the schemas section to describe the structure of a message. Under the publish operation of myChannel, you reference the MyMessageSchema component using the $ref keyword.
```

Comment:
I'm wondering if we should be applying styling to code objects/items here, such as `MyMessageSchema`, `myChannel`, etc. 

Thoughts, @derberg?